doc/src/parts/beginners_guide.rst

```diff
@@ -0,0 +1,482 @@
+and calculate how a difference in the x direction would translate to the y direction.
+Unfortunately this approach is not perfect though.
+Since we're extrapolating the derivative at the x data values, we will only receive valid results
+if the derivative doesn't change too much at the scale of the x error.
+Also, since the effective y error has now become dependent on the derivative of the model function
+it will vary depending on our choice of model parameters.
+This distorts our likelihood function - the minimum of a chi2 cost function will no longer be
+shaped like a parabola (with a model parameter on the x axis and chi2 on the y axis).
+
+To demonstrate this, the second file ``x_errors`` will perform a line-fit with much bigger
+uncertainties on the x axis than on the y axis. The non parabolic shape can be seen in the one
+dimensional profile scans.
+
+.. figure:: ../_static/img/004_x_errors_contours.png
+    :alt: Profiles and contour of a line fit with big x errors.
+
+kafe2go
+-------
+
+To display asymmetric parameter uncertainties use the flag ``-a``. In addition the profiles and
+contours can be shown by using the ``-c`` flag. In the *Python* example a ratio between the data
+and model function is shown below the plot. This can be done by appending the ``-r`` flag to
+*kafe2go*.
+
+.. bootstrap_collapsible::
+    :control_type: link
+    :control_text: non_linear_fit.yml
+
+    .. literalinclude:: ../../../examples/004_non_linear_fit/non_linear_fit.yml
+
+The dataset used to show that big uncertainties on the x axis can cause the fit to be non linear
+follows here.
+Keep in mind, that *kafe2go* will perform a line fit if no fit function has been specified.
+In order do add a grid to the contours, run *kafe2go* with the ``--grid all`` flag.
+So to plot with asymmetric errors, the profiles and contour as well as a grid run
+``kafe2go x_errors.yml -a -c --grid all``
+
+.. bootstrap_collapsible::
+    :control_type: link
+    :control_text: x_errors.yml
+
+    .. literalinclude:: ../../../examples/004_non_linear_fit/x_errors.yml
+
+Python
+------
+
+The according lines to display asymmetric uncertainties and to create the contour plot are
+highlighted in the code example below.
+
+.. bootstrap_collapsible::
+    :control_type: link
+    :control_text: non_linear_fit.py
+
+    .. literalinclude:: ../../../examples/004_non_linear_fit/non_linear_fit.py
+        :lines: 27-
+        :emphasize-lines: 23, 27, 29-31
+
+The example to show that uncertainties on the x axis can cause a non linear fit uses the *YAML*
+dataset given in the *kafe2go* section.
+
+.. bootstrap_collapsible::
+    :control_type: link
+    :control_text: non_linear_fit.py
+
+    .. literalinclude:: ../../../examples/004_non_linear_fit/x_errors.py
+        :lines: 19-
+
+
+Example 5: Histogram Fit
+========================
+
+*kafe2* is not only capable of performing XY-Fits. This example shows how to handle a one
```

Comment:
This sounds weird.
How about:
In addition to the xy fits shown above *kafe2* can also perform fits on one-dimensional datasets by binning the data.